mautrix/util/ffmpeg.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2021 Tulir Asokan
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from __future__ import annotations
+
+from typing import Iterable
+from pathlib import Path
+import asyncio
+import mimetypes
+import os
+import shutil
+import tempfile
+
+try:
+    from . import magic
+except ImportError:
+    magic = None
+
+
+def _abswhich(program: str | None) -> str | None:
+    path = shutil.which(program)
```

Comment:
This won't work if `program` is `None`. Doesn't look like there's any situation where we are passing `None` here, so it may not make any sense to have `| None` on the parameter.
```suggestion
    if program is None:
        return None
    path = shutil.which(program)
```